pandas/tests/indexes/test_base.py

```diff
@@ -803,12 +813,19 @@ def test_symmetric_difference(self):
         self.assertTrue(tm.equalContents(result, expected))
 
         # nans:
-        # GH #6444, sorting of nans. Make sure the number of nans is right
-        # and the correct non-nan values are there. punt on sorting.
         idx1 = Index([1, 2, 3, np.nan])
         idx2 = Index([0, 1, np.nan])
+        idx3 = Index([0, 1])
+
+        # GH 13514 change: {nan} - {nan} == {}
         result = idx1.symmetric_difference(idx2)
-        # expected = Index([0.0, np.nan, 2.0, 3.0, np.nan])
+        expected = Index([0.0, 2.0, 3.0])
+        tm.assert_index_equal(result, expected)
+
+        # GH #6444, sorting of nans. Make sure the number of nans is right
+        # and the correct non-nan values are there. punt on sorting.
+        result = idx1.symmetric_difference(idx3)
+        # expected = Index([0.0, 2.0, 3.0, np.nan])
```

Comment:
The lines testing #6444 were there before. I think I only changed the commented line with `expected`.

Though, it's very likely the issue with nan ordering no longer exists here. IIRC it's related with python set operations. I'll check if python 2 and 3 differ here and clean it if all is ok.